docs/release.md

```diff
@@ -0,0 +1,78 @@
+# Release
+
+> [!NOTE]
+> This document is based on the initial learnings from the release process and is therefore subject to change.
+
+## Preparing the release
+
+For each release, a milestone needs to be in place. All PRs that are part of the release need to be assigned to 
+the milestone.
+
+This will also make it easier to filter for the correct commits, that should be cherry-picked later.
+
+### Create an issue
+
+To track the progress of the release, we need to create an issue. Make use of the `Release` issue template.
+
+### Patch release
+
+A patch release can only include changes that are backwards compatible with the previous release. 
+This includes bug fixes, documentation updates, and other minor changes.
```

Comment:
I'd remove the "backwards compatible" thing here. It's true, but a bit misleading, as new features can also be backwards compatible, but still require a minor release.

Reply:
One additional  thing I missed previously. Talking about "**minor** changes" in the section for **patch** releases is also a bit misleading :)

Reply:
Yeah but I think this is understandable from the scope, because I don't know how to phrase it better. Except you have a good idea.

Reply:
"trivial" maybe?

Reply:
sold